myia/abstract/prim.py

```diff
@@ -86,7 +89,7 @@ def __init__(self, infer):
 def standard_prim(*prims):
     """Decorator to define and register a StandardInferrer."""
     def deco(fn):
-        xinf = StandardInferrer.partial(infer=fn)
+        xinf = StandardInferrer.partial(prim=prims[0], infer=fn)
```

Comment:
As far as I can see the only spot that uses multiple primitives is `P.identity` and `P.return_`.  Maybe that could just be some special case and we should enforce single-argument here.

All in all this doesn't make a great difference since it's almost always use for a single primitive, but the code here just feels wrong at first glance.

Reply:
Yeah, fair point, I'll enforce single argument.

Reply:
I made the change.